### ISSUE TITLE
[ASR Training] Loss nan during training

### ISSUE THREAD

**Issue by shihabtechno** [NVIDIA/NeMo#6033]
**W0215 06:48:41.808522 140192834852672 x2num.py:14] NaN or Inf found in input tensor**

**Pytorch Lighting Parameters**

omit amp_level and amp_backend,only used precision=16 to enable AMP

#amp_level="O1",
precision=16, 
#amp_backend='apex'

though it enables AMP [Using 16bit native Automatic Mixed Precision (AMP)],during training it shows training loss nan with a warning message **W0215 06:48:41.808522 140192834852672 x2num.py:14] NaN or Inf found in input tensor**

**Learning Rate =2**

![image](https://user-images.githubusercontent.com/107035118/218989436-147d27a4-78c5-4d20-a422-ca77d7adb645.png)

**Additional context**
- Nemo Docker version nemo:22.12
- Multi GPU

**Comment by titu1994** [comment]
Does the loss remain nan after the first few epochs ? AMP has an auto scalar in it that will scale loss and grads by some fixed integer / float power of 2 for a few dozen steps. This is expected to cause nan/inf and is fine, AMP is trying to find the highest scalar value that the model can tolerate without nan / inf. See https://pytorch.org/docs/stable/notes/amp_examples.html#cuda-automatic-mixed-precision-examples

**Comment by titu1994** [comment]
Which model and loss is being trained ? If it's conformer then the model may diverge unless you have a sufficiently large batch size

**Comment by titu1994** [comment]
You can use gradient accumulation to get global batch size to 256 or 128 to see if it stabilizes training
